### DISCUSSION TITLE
Hardware Requirements (and Raspberry Pi)

### DISCUSSION THREAD

**Discussion by pirachute** [asreview/asreview#1522] [Q&A]
Hi,

I am having some difficulty finding the Software Requirements for ASReview... I know probably to just run the application it doesn't require too much... but it would be interesting to add in the FAQ section some info about it and what kind of performance to expect.

Specificaly, as this can run as a server, it would be nice to know if we can make it run in a Raspberry Pi.

I don't know how resource intensive is to run this learning app for a computer and if it makes sense to do it on a low power computer, or if there's any benefit on using a powerful computer with a good video card...

Anybody can give some light on this topic?

**Comment by Rensvandeschoot** [comment]
Great idea!! We also recently started with a repository about running ASReview in the cloud:; https://github.com/asreview/cloud-usage

**Comment by jteijema** [comment]
Hi @pirachute,

Thank you for your interest in ASReview. I've successfully deployed ASReview on a Raspberry Pi 4 Model B with 4GB RAM, running Docker. The setup has also been tested on single-core instances with 512MB RAM.

We've chosen not to specify rigid hardware requirements because they can vary significantly depending on the model and configuration you choose. For example, using larger SBERT models can easily consume more than 2GB of RAM, whereas simpler methods like TF-IDF are far less demanding. Additionally, some classifiers take advantage of multi-threading, while others do not.

Could you share more details about your intended setup? Knowing more about your requirements would help us provide more targeted advice.

Best regards,
Jelle

**Comment by pirachute** [comment]
Hi!

I'm just  planning to use it for a simple literature review, nothing too complex... I personally have an Alienware with an I7 a very nice NVIDIA video card... but as I'm working on a shared project I was thinking about installing ASReview on a Raspberry Pi server so other people could access from different computers to review the work.

I just wanted to understand if there were any downsides for installing it in a Raspberry Pi or any advantages in using a more powerful computer in terms of performance... I know AI applications can be quite resource intensive and sometimes they require a video card to work...

Without being rigid, I think a "recommended" hardware or a couple of guidelines for the minimum requirements and expected performance could be helpful to encourage people to try this kind of setups.

Thank you very much for the project!! It's amazing and very well developed.

**Reply by Rensvandeschoot** [reply]
Thank you for the compliments! 
In terms of performance, you might want to look at this table: https://www.frontiersin.org/files/Articles/1178181/frma-08-1178181-HTML-r1/image_m/frma-08-1178181-t001.jpg. It shows the computation time for the feature extraction versus the computation time to run the classifier. The choice of the classifiers has the most impact on the run time, but this is just needed once. For the classifier, it does not really matter how long the training takes because while the machine trains a new model, you can continue screening (see this paper: https://psyarxiv.com/g93zf/).

**Reply by Rensvandeschoot** [reply]
For collaborating you might want to read more about the MEGA-meta project: https://asreview.nl/project/the-megameta-project/
